src/swap/v2/router/swap-router.ts

```diff
@@ -0,0 +1,203 @@
+import algosdk, {ALGORAND_MIN_TX_FEE, getApplicationAddress, Transaction} from "algosdk";
+import AlgodClient from "algosdk/dist/types/src/client/v2/algod/algod";
+
+import {CONTRACT_VERSION} from "../../../contract/constants";
+import {ALGO_ASSET_ID} from "../../../util/asset/assetConstants";
+import {getAssetId, isAlgo} from "../../../util/asset/assetUtils";
+import {SupportedNetwork} from "../../../util/commonTypes";
+import {getValidatorAppID} from "../../../validator";
+import {SwapType} from "../../constants";
+import {
+  FetchSwapRouteQuotesPayload,
+  FetchSwapRouteQuotesResponse,
+  GenerateSwapRouterTxnsParams
+} from "../../types";
+import {
+  V2_SWAP_APP_CALL_ARG_ENCODED,
+  V2_SWAP_APP_CALL_SWAP_TYPE_ARGS_ENCODED,
+  V2_SWAP_ROUTER_APP_ARGS_ENCODED
+} from "../constants";
+import {SWAP_ROUTER_INNER_TXN_COUNT, TINYMAN_ANALYTICS_API_BASE_URLS} from "./constants";
+import {
+  getAssetInFromSwapRoute,
+  getAssetOutFromSwapRoute,
+  getSwapRouterAppID
+} from "./util";
+
+export async function generateSwapRouterAssetOptInTransaction({
+  client,
+  routerAppID,
+  assetIDs,
+  accountAddress
+}: {
+  client: AlgodClient;
+  routerAppID: number;
+  assetIDs: number[];
+  accountAddress: string;
+}): Promise<Transaction> {
+  const suggestedParams = await client.getTransactionParams().do();
+  // We need to create a NoOp transaction to opt-in to the assets
+  const assetOptInTxn = algosdk.makeApplicationNoOpTxnFromObject({
+    from: accountAddress,
+    appIndex: routerAppID,
+    appArgs: [V2_SWAP_ROUTER_APP_ARGS_ENCODED.ASSET_OPT_IN],
+    foreignAssets: assetIDs,
+    suggestedParams
+  });
+  // The number of inner transactions is the number of assets we're opting in to
+  const innerTransactionCount = assetIDs.length;
+
+  /**
+   * The opt-in transaction fee should cover the total cost of inner transactions,
+   * and the outer transaction (thus the +1)
+   */
+  assetOptInTxn.fee = ALGORAND_MIN_TX_FEE * (innerTransactionCount + 1);
+
+  return assetOptInTxn;
+}
+
+export async function generateSwapRouterTxns({
+  initiatorAddr,
+  client,
+  network,
+  swapType,
+  route
+}: GenerateSwapRouterTxnsParams) {
```

Comment:
Let's just define it here. It serves like a documentation and it is only being used here
```suggestion
}: {
  client: Algodv2;
  initiatorAddr: string;
  swapType: SwapType;
  route: SwapRoute;
  network: SupportedNetwork;
}) {
```